homeworks/08-frp/src/index.ts

```diff
@@ -0,0 +1,71 @@
+import { Point } from "./point";
+import { GameCanvas } from "./canvas";
+import "./index.css";
+import {
+  getRandomInteger,
+  getWindowsPositions,
+  isNeighbours,
+} from "./calculations";
+import { fromEvent, interval } from "rxjs";
+import {
+  filter,
+  map,
+  scan,
+  startWith,
+  tap,
+  withLatestFrom,
+} from "rxjs/operators";
+import { GOAL_PICTURE, MAX, MIN } from "./constant";
+
+const scoreRender = (
+  score: number = 0,
+  scoreElement: HTMLElement | null
+): void => {
+  if (!!scoreElement) {
+    scoreElement.innerHTML = score.toString();
+  }
+};
+
+const main = () => {
+  const windowsCount = getRandomInteger(MIN, MAX);
+  const positions = getWindowsPositions(windowsCount);
+  const gameCanvas = new GameCanvas(positions, GOAL_PICTURE);
+  const scoreEl = document.getElementById("score");
+  const defaultPosition = positions[0];
+  gameCanvas.drawGameField();
+
+  const player$ = fromEvent(gameCanvas.canvas, "click").pipe(
+    map<MouseEvent, Point>((event) => ({ x: event.clientX, y: event.clientY }))
+  );
+
+  const goalPosition$ = interval(1000).pipe(
+    startWith(defaultPosition),
+    map(() => {
+      const positionIndex = getRandomInteger(1, windowsCount);
+      return positions[positionIndex];
+    }),
+    scan((acc, position) => {
+      gameCanvas.clearImage(acc);
```

Comment:
Bad practice to use any side-effects (like drawing on canvas) inside `scan` operator. Please try to split them into few separate operators.